### DISCUSSION TITLE
compdef is DEPRECATED

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [marlonrichert/zsh-snap#187] [Q&A] [answered]
According to the `znap help` compdef is DEPRECATED. Why is it and what is going to be the replacement. The current sample [.zshrc](https://github.com/marlonrichert/zsh-snap/blob/main/.zshrc) still uses it however.

**Comment by marlonrichert** [comment, marked as answer]
The example in the `.zshrc` file uses plain old `compdef`, not `znap compdef`.

It’s deprecated, because I decided there are better ways to do what it tried to accomplish. What you should use instead depends on what you want to accomplish. Can you give a particular example for which you now (are trying to) use `znap compdef`?

**Reply by Pilgrim1379** [reply]
No I have never used `znap compdef`. I just wanted to know your deprecating it and if there was a concrete replacement for those who may have relied on it.
